src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt

```diff
@@ -761,6 +776,7 @@ TOKEN : /* Data Types */
     | <#TYPE_REAL: "REAL" | "FLOAT4" | "FLOAT">
     | <#TYPE_DOUBLE: "DOUBLE" | "PRECISION"  | "FLOAT8" | "FLOAT64">
     | <#TYPE_VARCHAR: "NVARCHAR" | "VARCHAR" | "NCHAR" | <K_CHAR> | "BPCHAR" | "TEXT" | "STRING" | <K_CHARACTER> | "VARYING">
+    | <#TYPE_VARCHAR2: <K_VARCHAR2>>
```

Comment:
This should got to TYPE_VARCHAR since we try to establish a reasonable grouping please.

Reply:
This is a problem. When I move this into the `TYPE_VARCHAR`, the generated TokenManager gets a `code too large`. I guess we're reaching the limits of JavaCC again...

I'll have to look into why exactly this happens, but I've not applied this change yet in my last commit.